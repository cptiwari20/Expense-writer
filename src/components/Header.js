import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <nav>
    <h1>Expense Writer</h1>
    <ul>
      <li>
        <NavLink to='/'  activeStyle={{
          fontWeight: "bold",
          color: "red"
        }}
        exact={true}>
        Dashboard
        </NavLink>
      </li>
      <li>
        <NavLink to='/add-expense'  activeStyle={{
          fontWeight: "bold",
          color: "red"
        }}>
        Create
        </NavLink>
      </li>
      <li>
        <NavLink to='/about'  activeStyle={{
          fontWeight: "bold",
          color: "red"
        }}>
        About
        </NavLink>
      </li>
      <li>
      </li>
    </ul>
  </nav>
);

export default Header;