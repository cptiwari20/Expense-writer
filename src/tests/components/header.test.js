import React from "react";
// import ReactShallowRenderer from "react-test-renderer/shallow";

import { shallow } from 'enzyme';
import Header from "../../components/Header";

it('should render Header correctly', () => {
  const wrapper = shallow(<Header />);

  console.log(wrapper);
  
  // const renderer = new ReactShallowRenderer();
  // renderer.render(<Header />);

  // expect(renderer.getRenderOutput()).toMatchSnapshot();
})