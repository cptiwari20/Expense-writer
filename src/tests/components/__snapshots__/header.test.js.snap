// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<nav>
  <h1>
    Expense Writer
  </h1>
  <ul>
    <li>
      <NavLink
        activeClassName="active"
        activeStyle={
          Object {
            "color": "red",
            "fontWeight": "bold",
          }
        }
        aria-current="page"
        exact={true}
        to="/"
      >
        Dashboard
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="active"
        activeStyle={
          Object {
            "color": "red",
            "fontWeight": "bold",
          }
        }
        aria-current="page"
        to="/add-expense"
      >
        Create
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="active"
        activeStyle={
          Object {
            "color": "red",
            "fontWeight": "bold",
          }
        }
        aria-current="page"
        to="/about"
      >
        About
      </NavLink>
    </li>
    <li>
      <button
        onClick={[MockFunction]}
      >
        Log Out
      </button>
    </li>
  </ul>
</nav>
`;
